timestamp.go

```diff
@@ -1,18 +1,33 @@
 package zetasqlite
 
 import (
+	"fmt"
 	"strconv"
+	"strings"
 	"time"
-
-	"github.com/goccy/go-zetasqlite/internal"
 )
 
 // TimeFromTimestampValue zetasqlite returns string values ​​by default for timestamp values.
 // This function is a helper function to convert that value to time.Time type.
 func TimeFromTimestampValue(v string) (time.Time, error) {
-	f, err := strconv.ParseFloat(v, 64)
+	// ParseFloat is too imprecise to use, instead split into seconds and fractional seconds
+	parts := strings.Split(v, ".")
+	if len(parts) > 2 {
+		return time.Time{}, fmt.Errorf("invalid timestamp string (multiple delimiters) %s", v)
+	}
+	seconds, err := strconv.ParseInt(parts[0], 10, 64)
+	micros := int64(0)
```

Comment:
Since `err` variable may be overwritten by L26, so error handling should be done here.

```go
seconds, err := ...
if err != nil {
  return time.Time{}, err
}
micros := int64(0)
```